src/main/java/uk/gov/justice/digital/job/udf/JsonValidator.java

```diff
@@ -83,21 +83,21 @@ public static UserDefinedFunction createAndRegister(
             .register(
                 String.format("udfValidatorFor%s%s", schemaName, tableName),
                 udf(
-                    (UDF2<String, String, Boolean>) (String originalJson, String parsedJson) ->
+                    (UDF2<String, String, String>) (String originalJson, String parsedJson) ->
                         jsonValidator.validate(originalJson, parsedJson, schema),
-                        DataTypes.BooleanType
+                        DataTypes.StringType
                 )
             );
     }
 
-    public boolean validate(
+    public String validate(
```

Comment:
To enhance debugging of schema violations within the structured zone, an error message describing the validation result is used instead of a boolean. When the error message is empty then the record is considered valid.

Reply:
Does this cover of DPR-532 or is it only part of it ?

Reply:
It's only part of it.
At the moment resulting error would only be a single error summary e.g
`Record does not match schema nomis/offenders: some more detail here`
DPR-532 will be more detailed.